### ISSUE TITLE
Some Nerd Fonts don't work when Bolded or Italics

### ISSUE THREAD

**Issue by williamhCode** [kovidgoyal/kitty#5658]
**Describe the bug**
Some Nerd fonts don't get rendered if its bolded or italics.

**To Reproduce**
1. Set font to any Nerd Font
2. Go into vim and paste nerd font in ([some fonts here](https://github.com/ryanoasis/nerd-fonts/issues/144))

**Screenshots**
Normal:
<img width="573" alt="image" src="https://user-images.githubusercontent.com/83525937/201054319-c81ace83-c3fd-46cb-9e02-e47063fc3275.png">
Italics:
<img width="573" alt="image" src="https://user-images.githubusercontent.com/83525937/201054390-2559583b-b40f-4d75-9fb2-dfcf19448bf8.png">

**Environment details**
```
kitty 0.26.5 created by Kovid Goyal
Darwin Williams-MacBook-Pro-2.local 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:20:05 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T8101 arm64
ProductName:	macOS ProductVersion:	12.6 BuildVersion:	21G115
Frozen: True
Paths:
  kitty: /Applications/kitty.app/Contents/MacOS/kitty
  base dir: /Applications/kitty.app/Contents/Resources/kitty
  extensions dir: /Applications/kitty.app/Contents/Resources/Python/lib/kitty-extensions
  system shell: /opt/homebrew/bin/zsh
Loaded config files:
  /Users/williamhou/.config/kitty/kitty.conf

Config options different from defaults:
background_image        /Users/williamhou/Pictures/Wallpapers/dark-tree.png
background_image_layout scaled
background_tint         0.9
dim_opacity             0
font_family             Fira Code
font_size               17.0
hide_window_decorations 2
macos_option_as_alt     3
modify_font:
    cell_width 90%
Added shortcuts:
	cmd+/ →  send_text all \x1b/
	cmd+[ →  send_text all \x1b[
	cmd+] →  send_text all \x1b]
	cmd+b →  send_text all \x1bn
	cmd+backspace →  send_text all \x15
	cmd+j →  send_text all \x01z
	cmd+l →  send_text all \x01L
	ctrl+- →  send_text all \x1f
Changed shortcuts:
	cmd+r →  send_text all \x01s
Colors:
	background              #1a1d22   
	color0                  #282c34   
	color1                  #e06c75   
	color10                 #56b6c2   
	color11                 #e5c07b   
	color12                 #61afef   
	color13                 #be5046   
	color14                 #56b6c2   
	color15                 #abb2bf   
	color2                  #98c379   
	color3                  #e5c07b   
	color4                  #61afef   
	color5                  #be5046   
	color6                  #56b6c2   
	color7                  #979eab   
	color8                  #393e48   
	color9                  #d19a66   

Important environment variables seen by the kitty process:
	PATH                                /Applications/kitty.app/Contents/MacOS:/usr/bin:/bin:/usr/sbin:/sbin
	LANG                                en_US.UTF-8
	SHELL                               /opt/homebrew/bin/zsh
	DISPLAY                             /private/tmp/com.apple.launchd.qVoFCThUVa/org.xquartz:0
	USER                                williamhou

```

**Additional context**
Minimal Config:
```
font_family Fira Code
```
I tried this with iTerm2 and it works there. Occasionally all fonts also glitch out and I have to reopen a few times and resource config to get them working.

kitty version=0.26.5
macOS version=12.6

**Comment by kovidgoyal** [comment]
Don't use NERD fonts, use a normal font and use symbol_map for nerd symbols.
https://sw.kovidgoyal.net/kitty/faq/?highlight=nerd#kitty-is-not-able-to-use-my-favorite-font 

Or if you must use NERD fonts, then use one for italic as well in
kitty.conf.

And this issue is fundamentally because NERD fonts incorrectly repurpose
non Private Use Area Unicode code points. They need to fix that.

**Comment by williamhCode** [comment]
Ok thanks! Didn't know this before.

**Comment by williamhCode** [comment]
@kovidgoyal sorry for bothering, but I have a quick question. From the link you sent me, it says that kitty automatically picks up nerd font, which it does. The problem is that the italics thing happens, so I use symbols_map (from the code in the link). But what happens is that the symbols become slightly off centered and slightly different in size. Is that normal?

Without symbols_map:
<img width="257" alt="image" src="https://user-images.githubusercontent.com/83525937/201199270-e5ce37b8-1392-4d23-aa94-fdb2a562f511.png">
<img width="144" alt="image" src="https://user-images.githubusercontent.com/83525937/201199581-daecf680-a8f2-477e-81cb-2a748950a6a5.png">

With symbols_map:
<img width="152" alt="image" src="https://user-images.githubusercontent.com/83525937/201199719-4978126c-afb3-4a53-91f3-f8187479c584.png">
<img width="144" alt="image" src="https://user-images.githubusercontent.com/83525937/201199653-f6f560c6-939d-41d3-8bde-b76da36650b7.png">

**Comment by kovidgoyal** [comment]
It depends on the details of which font is used and how its metrics compare to the nerd font. And which nerd font version you are using, best is to use the plain nerd fonts font not a patched one.

**Comment by williamhCode** [comment]
I just realized that kitty is picking up my Hack Nerd Font automatically, which actually has the most correct symbols (centered and right size). Everything is fixed after I symbol mapped the Hack Nerd Font. They really need to fix their symbols only font idk why is different from the other nerd fonts. Thx for the help!

**Comment by David-Else** [comment]
@kovidgoyal I have followed your FAQ advice https://sw.kovidgoyal.net/kitty/faq/#kitty-is-not-able-to-use-my-favorite-font and installed `NerdFontsSymbolsOnly`. I am trying to use them in LazyGit, but it only works when I am selecting a line:

![Screenshot from 2022-11-30 23-08-04](https://user-images.githubusercontent.com/12832280/205020766-58b4c2b0-a091-4b8d-bd84-9537bdf1827d.png)

The only font that is correct is the highlighted one, the others are identical to when there is no `NerdFontsSymbolsOnly` installed. If I move the highlight then the font becomes correct only under the hightlight. It is like this all over the screen.

This has been reported to LazyGit maintainers https://github.com/jesseduffield/lazygit/issues/2236, but is it possible this is a Kitty problem? Can you shed any light on this unfortunate situation? Cheers!

**Comment by kovidgoyal** [comment]
Dont just install the font add symbol_map to kitty.conf as well.

**Comment by David-Else** [comment]
@kovidgoyal Thanks for the quick response above,  it almost works perfectly now. My only remaining problem after taking your advice is that one particular symbol is displaying wrong. If you look at the image I posted above, now it is always the wrong symbol, highlighting it changes nothing. 

I don't really understand what the following I copied does:
```
symbol_map U+23FB-U+23FE,U+2665,U+26A1,U+2B58,U+E000-U+E00A,U+E0A0-U+E0A3,U+E0B0-U+E0C8,U+E0CA,U+E0CC-U+E0D2,U+E0D4,U+E200-U+E2A9,U+E300-U+E3E3,U+E5FA-U+E634,U+E700-U+E7C5,U+EA60-U+EBEB,U+F000-U+F2E0,U+F300-U+F32F,U+F400-U+F4A9,U+F500-U+F8FF Symbols Nerd Font Mono
```
Could it be possible to change one of these values to get the right symbol back, I am not sure how to debug this?

Lazygit is meant to be nerd fort compatible, and it is apart from this correct symbol:
![rightpng](https://user-images.githubusercontent.com/12832280/205088863-4d7a387e-6848-49e8-8759-abc79ced617f.png)

is displayed as this:
![wrong](https://user-images.githubusercontent.com/12832280/205088870-b7b7349f-c86e-4f28-80ab-53942a108cdc.png)

When using the `symbol_map`.

I am using the exact same symbol fonts as your example:

![Screenshot from 2022-12-01 15-19-19](https://user-images.githubusercontent.com/12832280/205090471-6c2cea1e-b284-4312-ad3e-56f5b6411b5e.png)
